### ISSUE TITLE
Add Storybook story for datastore-driven admin bar menu

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2594]
After #2076, the admin bar menu is now (behind a feature flag) using the datastore. However, Storybook only has coverage for the old version. Let's add another story to `admibar.stories.js` that covers the new version of it. We need to remove the `disableFeature( 'widgets.adminBar' );` line from `resetFeatures` and instead handle this within the specific story: The legacy story should have the feature flag disabled and add legacy filters. The new story should have the feature flag enabled and not prevent any legacy filters from being added.

The main part of work here is to populate the datastore with appropriate mock data.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* There should be an additional story for the new version of the admin bar menu in `adminbar.stories.js`.
* The old admin bar story should be maintained, and it should not display the new parts of the admin bar.

## Implementation Brief
* Using `/.storybook/config.js`
	- Update the `resetFeatures` function to remove the line which disables the admin bar (widgets.adminBar)
* Using `/stories/adminbar.stories.js`,
	* Import `disableFeature` and `enableFeature` from `/stories/utils/features`
	* Update the "Admin Bar" story, to disable the new admin bar, using the `disableFeature` function. For e.g `disableFeature( 'widgets.adminBar' );`
	* Add a new story for the new datastore-driven admin bar, which renders the same output as the "Admin Bar" story but without the `CollectModuleData` component.
	* Using the newly added story,
		* Remove all existing filters and enable the admin bar via the `enableFeature` function.
			```js
			removeAllFilters( 'googlesitekit.AdminbarModules' );
			enableFeature( 'widgets.adminBar' );
			```
		* The `setupRegistry` function should do the following:
			* Activate the `search-console` module, which technically we can skip because search console is always connected and active.
			* Provide the site's info.
			* Set a reference data for the user.
			* Provide data and options to the module's store.
			* Use the `receiveGetReport` selector of the module's store to populate the data.
			* A working code snippet of the above bullet points [can be found here](https://gist.github.com/asvinb/0de95e766bdf379d485d93beadb99675).
		* Repeat the previous steps to populate the `analytics` data store, taking sample data from `/stories/module-analytics-components.stories.js` and adjusting the dates so that they match the same date range as the search console data.

### Test Coverage
* No changes expected.

### Visual Regression Changes
* No changes expected.

## QA Brief

* Run your local Storybook set up with `npm run storybook`
* Check that the [Admin Bar story](http://localhost:9001/?path=/story/global--admin-bar) renders correctly and the layout matches the existing version (Notes: the data will have changes as the mock data has been updated.).

## Changelog entry

* N/A

**Comment by felixarntz** [comment]
IB ✅

**Comment by benbowler** [comment]
I reversed the blocking here. Otherwise, we have to build out conditional logic to show the previous and new version of the admin bar in Storybook. It's simpler to switch to the new data-driven admin bar and then we only have to update Storybook to show the new admin bar.

**Comment by benbowler** [comment]
QA Brief added

**Comment by benbowler** [comment]
[Complete in this PR.](https://github.com/google/site-kit-wp/pull/2639)

**Comment by aaemnnosttv** [comment]
@felixarntz 

> We need to remove the `disableFeature( 'widgets.adminBar' );` line from `resetFeatures` and instead handle this within the specific story: The legacy story should have the feature flag disabled and add legacy filters.

Would you please elaborate on why this is needed? The point of `resetFeatures` is to give every story a clean slate before it runs. Every story should enable the features it needs explicitly rather than the reverse. Once #2452 is merged, this would be done by passing a new `features` prop to `WithTestRegistry` otherwise all extra features are disabled by default.

> The new story should have the feature flag enabled and not prevent any legacy filters from being added.

Similar to the previous comment, for stories that rely on filters those stories need to `removeAllFilters` first and then re-add them otherwise the story will accumulate callbacks to the filter every time the story runs.

**Comment by benbowler** [comment]
As this ticket itself resolved E2E and VRT issues with #2241 I've merged this branch down to that one in [PR 2630](https://github.com/google/site-kit-wp/pull/2630).

**Comment by tofumatt** [comment]
I’m a bit confused about how to handle this being in Code Review as the PR that addresses it has already been marked as merged… I know it was merged into another branch of yours.

Might be better to say #2630 resolves both issues and explain that in the PR. We tend not to have PRs for multiple issues, but in this case it makes more sense to review one PR I think. Otherwise we'll be reviewing the code in #2639 twice.

Unless the #2639 PR can be made against `develop`. 🙂

**Comment by benbowler** [comment]
In order to get #2241 through automated tests and CR I ended up merging the changes from #2594(this ticket)->#2241. Now #2241, the work in this ticket has now been approved and merged. The question is, what happens to this ticket? Should we close it as duplicate or move it to QA directly?

@tofumatt @aaemnnosttv

**Comment by tofumatt** [comment]
QA ✅
